### ISSUE TITLE
tf.keras.utils.multi_gpu_model does use only one GPU

### ISSUE THREAD

**Issue by dmus** [tensorflow/tensorflow#21788]
Tensorflow 1.10
Docker container 1.10.0-gpu-py3 from https://hub.docker.com/r/tensorflow/tensorflow/
I think this is a bug. I expect tf.keras.utils.multi_gpu_model to use multiple GPUs. However with the following example it uses only 1 (checking with nvidia-smi):

```
import tensorflow as tf
from tensorflow.keras.applications import Xception
from tensorflow.keras.utils import multi_gpu_model
import numpy as np

num_samples = 32
height = 224
width = 224
num_classes = 1000

with tf.device('/cpu:0'):
    model = Xception(weights=None,input_shape=(height, width, 3),classes=num_classes)

# Replicates the model on 2 GPUs.
# This assumes that your machine has 2 available GPUs.
parallel_model = multi_gpu_model(model, gpus=2)
parallel_model.compile(loss='categorical_crossentropy', optimizer='rmsprop')

# Generate dummy data.
x = np.random.random((num_samples, height, width, 3)).astype(np.float32)

dataset = tf.data.Dataset.from_tensor_slices(x)
dataset = dataset.batch(32)
dataset = dataset.repeat()

iterator = dataset.make_one_shot_iterator()
batch = iterator.get_next()

y = parallel_model(batch)

sess = tf.keras.backend.get_session()
while True:
    try:
        result = sess.run(y)
        print(result.shape)
    except tf.errors.OutOfRangeError:
        break
```

See also https://stackoverflow.com/questions/51962659/tf-keras-utils-multi-gpu-model-does-use-only-one-gpu

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by dmus** [comment]
@robieta what do you think?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by dmus** [comment]
Testing again with tensorflow/tensorflow:1.12.0-rc2-gpu-py3

This issue is not solved yet, still only 1 GPU is used.

Output of nvidia-smi looks like this:
![screenshot from 2018-11-01 12-54-49](https://user-images.githubusercontent.com/464378/47850572-fe5bc600-ddd5-11e8-8829-fee5bd227fc4.png)

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by robieta** [comment]
I replaced the last section of your code
```
iterator = dataset.make_one_shot_iterator()
batch = iterator.get_next()

y = parallel_model(batch)

sess = tf.keras.backend.get_session()
while True:
    try:
        result = sess.run(y)
        print(result.shape)
    except tf.errors.OutOfRangeError:
        break
```

with

```
parallel_model.fit(dataset, steps_per_epoch=1000)
```

And it parallelized as expected.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by MonlinXING** [comment]
I'm still getting this problem,even at the begining before the actual computing. The memory usage of one GPU is higher than the others. What could be the cause of this problem? @robieta Thank you.
![image](https://user-images.githubusercontent.com/30249811/60592533-6a625400-9dd3-11e9-83b7-ff8b44f9aa79.png)

**Comment by weichen456** [comment]
I have the same problem,

**Comment by SuperAcidLemon** [comment]
I have the same issue.

**Comment by MosheSharabi** [comment]
any solution for it ? i cant use tf.distribute.Strategy because i use fit_generator, and disable eager mode dont help too

**Comment by kylekyle** [comment]
Same issue.
